sql/plan/transformed_named_node.go

```diff
@@ -0,0 +1,55 @@
+// Copyright 2020-2021 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package plan
+
+import (
+	"github.com/dolthub/go-mysql-server/sql"
+)
+
+type TransformedNamedNode struct {
```

Comment:
Probably could use a comment here, something like

TransformedNamedNode is a wrapper for arbitrary logic to represent a table factor assembled from other nodes

See e.g. Concat